Simplified/Reader2/ReaderStackConfiguration/Axis/NYPLAxisProtectedAssetHandler.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  NYPLAxisProtectedAssetHandler.swift
+//  Open eBooks
+//
+//  Created by Raman Singh on 2021-05-18.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+import Foundation
+import R2Shared
+import R2Streamer
+
+
+typealias ProtectedAssetCompletion = (Result<ProtectedAsset, Publication.OpeningError>) -> Void
+
+protocol NYPLAxisProtectedAssetHandling {
+  func handleAsset(asset: FileAsset, fetcher: Fetcher,
+                   completion: @escaping ProtectedAssetCompletion)
+}
+
+struct NYPLAxisProtectedAssetHandler: NYPLAxisProtectedAssetHandling {
+  
+  let axisKeysProvider: NYPLAxisKeysProviding
+  let decryptor: NYPLAxisContentDecrypting
+  let licenseDownloader: NYPLAxisItemDownloading
+  
+  init(
+    axisKeysProvider: NYPLAxisKeysProviding = NYPLAxisKeysProvider(),
+    decryptor: NYPLAxisContentDecrypting,
+    licenseDownloader: NYPLAxisItemDownloading = NYPLAxisItemDownloader(
+      dispatchGroup: DispatchGroup())) {
+    
+    self.axisKeysProvider = axisKeysProvider
+    self.decryptor = decryptor
+    self.licenseDownloader = licenseDownloader
+  }
+  
+  init?(axisKeysProvider: NYPLAxisKeysProviding = NYPLAxisKeysProvider(),
+        decryptor: NYPLAxisContentDecrypting? = NYPLAxisContentDecryptor(),
+        licenseDownloader: NYPLAxisItemDownloading = NYPLAxisItemDownloader(
+          dispatchGroup: DispatchGroup())) {
+    
+    guard let decryptor = decryptor else {
+      return nil
```

Comment:
this seems a little odd: you have an initializer where you're providing a default Decryptor, so why not using the default decryptor if the caller does not provide one? I may be missing something so feel free to disagree.

Reply:
NYPLAxisContentDecryptor() has a failable initializer because it depends on NYPLRSACypher which itself has a failable initializer. Another way to do this was initiating if after making sure we have a cypher object. I chose this because it makes the code more readable.